genesis/engine/solvers/fem_solver.py

```diff
@@ -265,6 +267,63 @@ def init_surface_fields(self):
             layout=ti.Layout.SOA,
         )
 
+    def compute_surface_info(self):
+        self.vertices_on_surface = ti.field(dtype=ti.u1, shape=(self.n_vertices))
+        self.elements_on_surface = ti.field(dtype=ti.u1, shape=(self.n_elements))
+        self.compute_surface_vertices()
+        self.compute_surface_elements()
+        self.vertices_on_surface_np = self.vertices_on_surface.to_numpy()
+        self.elements_on_surface_np = self.elements_on_surface.to_numpy()
+        self.surface_vertices_np = np.where(self.vertices_on_surface_np)[0].flatten()
+        self.surface_vertices = ti.field(
+            dtype=ti.i32,
+            shape=(self.surface_vertices_np.shape[0]),
+            needs_grad=False,
+        )
+        self.surface_vertices.from_numpy(self.surface_vertices_np)
+        self.surface_elements_np = np.where(self.elements_on_surface_np)[0].flatten()
+        self.surface_elements = ti.field(
+            dtype=ti.i32,
+            shape=(self.surface_elements_np.shape[0]),
+            needs_grad=False,
+        )
+        self.surface_elements.from_numpy(self.surface_elements_np)
+
+        self.surface_triangles_np = self.surface.tri2v.to_numpy().reshape(-1, 3)
+        pos_np = self.elements_v.pos.to_numpy()[0, :, 0, :].reshape(-1, 3)[self.surface_vertices_np]
+        surface_vertices_mapping = np.ones(self.n_vertices, dtype=np.int32) * -1
+        for i, v in enumerate(self.surface_vertices_np):
+            surface_vertices_mapping[v] = i
+        mass = igl.massmatrix(pos_np, surface_vertices_mapping[self.surface_triangles_np])
+        self.surface_vert_mass_np = mass.diagonal()
+        self.surface_vert_mass = ti.field(
+            dtype=gs.ti_float,
+            shape=(self.surface_vertices_np.shape[0]),
```

Comment:
Just do `(len(surface_vertices_np),)` instead of `.shape[0]`, plus you forgot the trailing comma here.